src/main/java/com/projectkorra/rpg/modules/worldevents/schedule/util/ScheduleParser.java

```diff
@@ -0,0 +1,53 @@
+package com.projectkorra.rpg.modules.worldevents.schedule.util;
+
+import java.time.Duration;
+import java.time.LocalTime;
+import java.util.Locale;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public final class ScheduleParser {
+    private static final Pattern TIME = Pattern.compile("(\\d{1,2})(?::(\\d{2}))?(am|pm)?", Pattern.CASE_INSENSITIVE);
+    private static final Pattern DURATION = Pattern.compile("(\\d+)([dhms])", Pattern.CASE_INSENSITIVE);
+
+    private ScheduleParser() {}
+
+    public static LocalTime parseTimeOfDay(String raw, LocalTime fallback) {
+        if (raw == null || raw.isBlank()) return fallback;
+        String s = raw.trim().toLowerCase(Locale.ROOT);
+        Matcher m = TIME.matcher(s);
+        if (!m.matches()) return fallback;
+
+        int hour = Integer.parseInt(m.group(1));
+        int minute = m.group(2) != null ? Integer.parseInt(m.group(2)) : 0;
+        String period = m.group(3);
+
+        if (period != null) {
+            if ("pm".equals(period) && hour < 12) hour += 12;
```

Comment:
Might be a personal thing but could we swap the direction of these lol? I get the "what if period is null" reasoning but you have a null check right above it. I just think it looks better and is more readable the other way around + the reasoning for swapping it this way is already handled in your implementation.